lib/Parser/Parse.cpp

```diff
@@ -9299,8 +9313,26 @@ ParseNodePtr Parser::ParseExpr(int oplMin,
 
             // Parse the operand, make a new node, and look for more
             IdentToken token;
+            bool coalescing = false;
+
             ParseNode* pnode2 = ParseExpr<buildAST>(
-                opl, nullptr, fAllowIn, FALSE, pNameHint, &hintLength, &hintOffset, &token, false, nullptr, plastRParen);
+                opl, nullptr, fAllowIn, FALSE, pNameHint, &hintLength, &hintOffset, &token, false, nullptr, plastRParen, &coalescing);
+
+            if (nop == knopLogAnd || nop == knopLogOr)
+            {
+                if (localCoalesce || coalescing)
```

Comment:
Here's the check for the syntax error - `localCoalesce` will be true if the left hand side of this operation is an unbracketed ??,  `coalescing` will be true if the right hand side is.

Aware the variable names aren't the best - open to alternatives.